theano/gof/destroyhandler.py

```diff
@@ -808,13 +808,20 @@ def fast_destroy(self, app):
             destroyed_inputs_idx += i
         for d_inp_idx in destroyed_inputs_idx:
             d_inp = app.inputs[d_inp_idx]  # The destroyed input variable
-            if len(d_inp.clients) == 1:
-                # We are fine, the algo tell we shouldn't generate cycle
-                continue
-            # Current simplest version don't allow destroyed inputs to
-            # have more then 1 client.
+            if len(d_inp.clients) != 1:
+                # We allow only 1 clients on variable we destroy
                 self.fail_validate = True
-                
+
+            # If the variable destroyed isn't a view, we are fine.
+            if not d_inp.owner or not getattr(d_inp.owner.op,
+                                              'view_map', False):
+                continue
+            # The node have views, verify if this specific output is a view.
```

Comment:
One small clarification. If a variable doesn't have an owner, it is an input right ?